### PR TITLE
Commit update Coin98 Wallet Adapter

### PR DESCRIPTION
**Pull request by khiemsoft**

### CODE REVIEW

**Comment by khiemsoft** [comment]
New Updated Coin98 Wallet Adapter
- Implement BaseMessageSignerWalletAdapter
- Support method signMessage
- Update method signAllTransactions
- Change icon to C98 Wallet Logo

File: packages/core/base/src/adapter.ts [hunk 1]
```diff
@@ -106,14 +106,24 @@ export function scopePollingDetectionStrategy(detect: () => boolean): void {
...
     }
 
-    function listener() {
-        window.removeEventListener('load', listener);
+    function windowLoadListener() {
+        window.removeEventListener('load', windowLoadListener);
         poll();
     }
-    window.addEventListener('load', listener);
+    window.addEventListener('load', windowLoadListener);
```
Comment (jordansexton) [inline packages/core/base/src/adapter.ts:128]: This was done in #348! Please revert these changes :)

File: packages/wallets/coin98/src/adapter.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ interface Coin98Wallet {
     request(params: { method: string; params: string | string[] | unknown }): Promise<{
         signature: string;
         publicKey: string;
+        sig?: string;
```
Comment (jordansexton) [inline packages/wallets/coin98/src/adapter.ts:24]: Why is there a `sig` and `signature` field? Why not just use the `signature` field?

File: packages/wallets/coin98/src/adapter.ts [hunk 3]
```diff
@@ -36,11 +37,10 @@ export interface Coin98WalletAdapterConfig {}
...
...Hn13E6NLJ5RTxlUygqdE0wTY3VbS5EYPBaSay7tgQ49HKIE+ejnDgf4+LVObY86Wd1uwu3UyOTsx8OiFIKp6nRXG8iwORskem4hWCBCB6XxmPNTmwFM7Eiur4gcqWgsc5kTYeTlpDjnvcKXRekygFYBQjU+XV+eSCE163x6lszfDaexdAFy1Zs6vHw0wON3I0V+c1rk9yeLgCwvtuD3xvj74NxHmtz8t7FGE5T2LTOi64Lk9N5shXYgCrOKgLxlE00aRGqMXh5ew2tDSamIXS1ONm/I0AwoDM1WyCRshgcThFPWWzq9XBoXwOmqfHG8WlyeZvv721g5+YAmazNpatzFKocfJU1IpDO2ZweTLCm1cnWPg8djQ5mEhatIZPWkIPpWJH3LiewbLg2nuHkBzG+t6eeA7uDPLvRz91okVCdSWe7C90QTl2I8dGnc2hfx0cANBEmI0XiKYv2RgetDSbtjQ7cLo3wVIHXT0X4ZDRTsnVVMrZsXtEactDa6GBVq5PagEk0UeSdc1GOHp8hnbG/Kt3FPgLS09OjKgFZaAvNQZPeVS5qfDrRpMXoRI6ZmLVkURGhrdFBb6eb2oBBMmMzFs4xMZkv/eVVuBN9lW85IPOHVfnHD0TQtBJjlS5YS8aWx1e7cC2rkfvFW0pMxUUWqX8JSywbIoIGZJd0qGqzV+xCkdWA0UeXr3IopW5owFGg+JBTVgKHBRzVgD8D794H8VESchL4kwFEgF8ANrAfMB/Ex/9x1wplEL8CIoaUEtwCfggcBl4BWUvpKrqYmgcgeUiQAuSAUaX4qwhvAFGA/wAjUJfreOScGwAAADhlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAAqACAAQAAAABAAABLKADAAQAAAABAAABLAAAAAD7qKDdAAAAJXRFWHRkYXRlOmNyZWF0ZQAyMDIyLTAyLTI4VDAzOjU3OjMzKzAwOjAwYHFggAAAACV0RVh0ZGF0ZTptb2RpZnkAMjAyMi0wMi0yOFQwMzo1NzozMyswMDowMBEs2DwAAAASdEVYdGV4aWY6RXhpZk9mZnNldAAyNlMbomUAAAAYdEVYdGV4aWY6UGl4ZWxYRGltZW5zaW9uADMwMEW2lAMAAAAYdEVYdGV4aWY6UGl4ZWxZRGltZW5zaW9uADMwMNi5dXUAAAAASUVORK5CYII=';
```
Comment (jordansexton) [inline packages/wallets/coin98/src/adapter.ts:43]: Any chance you have an original SVG instead of a PNG we can use? It should be much smaller.

File: packages/wallets/coin98/src/adapter.ts [hunk 4]
```diff
@@ -158,7 +158,37 @@ export class Coin98WalletAdapter extends BaseSignerWalletAdapter {
         const signedTransactions: Transaction[] = [];
         for (const transaction of transactions) {
             signedTransactions.push(await this.signTransaction(transaction));
+            await this.sleep()
```
Comment (jordansexton) [inline packages/wallets/coin98/src/adapter.ts:161]: ... What is this for?

File: packages/wallets/coin98/src/adapter.ts [hunk 5]
```diff
@@ -158,7 +158,37 @@ export class Coin98WalletAdapter extends BaseSignerWalletAdapter {
...
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        }
+    }
+
+    //Ultilities
+    sleep(ms = 500): Promise<boolean>{
+        return new Promise(resolve => setTimeout(resolve, ms))
+    }
```
Comment (jordansexton) [inline packages/wallets/coin98/src/adapter.ts:192]: Please remove, timeouts are not a reliable, deterministic source of async evaluation. I'm also not sure what the delay is for.

File: packages/wallets/coin98/src/adapter.ts [hunk 6]
```diff
@@ -158,7 +158,37 @@ export class Coin98WalletAdapter extends BaseSignerWalletAdapter {
...
     }
+
+
+    async signMessage(message: Uint8Array): Promise<Uint8Array> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                const decodedMessage = new TextDecoder().decode(message)
```
Comment (jordansexton) [inline packages/wallets/coin98/src/adapter.ts:173]: Hmm it would probably be better to standardize your interface around something common like Base64 than using `TextDecoder.decode`'s return value of a [DOMString](https://developer.mozilla.org/en-US/docs/Web/API/DOMString).

File: packages/wallets/coin98/src/adapter.ts [hunk 7]
```diff
@@ -158,7 +158,37 @@ export class Coin98WalletAdapter extends BaseSignerWalletAdapter {
...
+                
+                const response = await wallet.request({ method: 'sol_sign', params: [decodedMessage] });
+
+                const signature = new TextEncoder().encode(response.sig);
```
Comment (jordansexton) [inline packages/wallets/coin98/src/adapter.ts:177]: Same here, can we use `response.signature` and Base64?

**Review by jordansexton — CHANGES_REQUESTED** [review]
Thanks for this! Much easier to review after the rebase.
I have a few change requests and a few questions.

File: packages/wallets/coin98/src/adapter.ts (hunk 6 above)
Comment (jordansexton) [reply packages/wallets/coin98/src/adapter.ts:173]: Oh, base58 might be better, as that's what's used for standard Solana transaction signatures.

File: packages/wallets/coin98/src/adapter.ts (hunk 7 above)
Comment (jordansexton) [reply packages/wallets/coin98/src/adapter.ts:177]: Oh, base58 might be better, as that's what's used for standard Solana transaction signatures.

**Comment by khiemsoft** [comment]
Fixed your comment on Coin98 Wallet Adapter

**Review by jordansexton — COMMENTED** [review]
Looks good! I only had one question about a possible typo in the method being called.

File: packages/wallets/coin98/src/adapter.ts [hunk 8]
```diff
@@ -155,10 +155,46 @@ export class Coin98WalletAdapter extends BaseSignerWalletAdapter {
...
-        for (const transaction of transactions) {
-            signedTransactions.push(await this.signTransaction(transaction));
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                const response = await wallet.request({ method: 'sol_signAllTransaction', params: [transactions] });
+                const publicKey = new PublicKey(response.publicKey);
+                const signatures = response.signature
```
Comment (jordansexton) [inline packages/wallets/coin98/src/adapter.ts:165]: Actually, this too. Shouldn't this be `response.signatures`?

File: packages/wallets/coin98/src/adapter.ts [hunk 9]
```diff
@@ -155,10 +155,46 @@ export class Coin98WalletAdapter extends BaseSignerWalletAdapter {
...
     async signAllTransactions(transactions: Transaction[]): Promise<Transaction[]> {
-        const signedTransactions: Transaction[] = [];
-        for (const transaction of transactions) {
-            signedTransactions.push(await this.signTransaction(transaction));
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                const response = await wallet.request({ method: 'sol_signAllTransaction', params: [transactions] });
```
[the remark this answers is not in the document]
Comment (khiemsoft) [reply packages/wallets/coin98/src/adapter.ts]: Comment Fixed

File: packages/wallets/coin98/src/adapter.ts (hunk 8 above)
Comment (khiemsoft) [reply packages/wallets/coin98/src/adapter.ts]: Comment Fixed
